src/protocol_versions.json

```diff
@@ -1,22 +1,26 @@
 {
-    "1.0.0": {
-        "min": "0.1.0",
-        "max": "0.1.0"
-    },
-    "1.0.1": {
-        "min": "0.1.0",
-        "max": "0.1.0"
-    },
-    "1.0.2": {
-        "min": "0.1.0",
-        "max": "0.1.0"
-    },
-    "1.0.3": {
-        "min": "0.1.0",
-        "max": "0.1.0"
-    },
-    "1.0.4": {
-        "min": "0.1.0",
-        "max": "0.1.0"
-    }
+  "1.0.0": {
+    "min": "0.1.0",
+    "max": "0.1.0"
+  },
+  "1.0.1": {
+    "min": "0.1.0",
+    "max": "0.1.0"
+  },
+  "1.0.2": {
+    "min": "0.1.0",
+    "max": "0.1.0"
+  },
+  "1.0.3": {
+    "min": "0.1.0",
+    "max": "0.1.0"
+  },
+  "1.0.4": {
+    "min": "0.1.0",
+    "max": "0.1.0"
+  },
+  "1.0.5": {
+    "min": "0.3.1",
+    "max": "0.3.1"
```

Comment:
What happens when backend blockchain node is restarted with ain-blockchain package upgraded to 0.3.2 but user still uses this package with version 1.0.5?

Reply:
Hmm how about we discard the patch numbers and only keep the major & minor versions? So this package currently supports protocol v0.3.x, and we would change on blockchain's side to only check the first two numbers, instead of matching the entire sequence. If the protocol version has another major/minor upgrades, ain-js should also update code and upgrade its major and/or minor version accordingly.

Reply:
My original question is whether we need to leave the 'max' part as a blank as we still don't know as of now (and this can be updated later once it's determined)?

Regarding the version resolution idea, basically I think it's a good idea. One option is:
- Whether it should be 0.3.1 or just 0.3 is determined and specified in this file
- If it's 0.3 rather than 0.3.0 then the compatibility checking logic would interpret that it's compatible with any 0.3.x. 
- Otherwise, e.g. 0.3.1, it's interpreted with only 0.3.1 (not 0.3.0 for example)

What do you think?

Reply:
I think your idea is quite straightforward, and I like that it gives us more control than always discarding the patch version. One thing I'm not too clear is how blockchain should manage the protocol versions & their compatibilities. Should we manage our versions so that all 0.3.x versions are compatible with each other (in which case we don't really need a matrix to look up), or should we have a version matrix similar to what we have right now?

Reply:
Updated the code as discussed offline and blockchain is also being updated (https://github.com/ainblockchain/ain-blockchain/pull/125) PTAL @platfowner